### PR TITLE
feat: migrate to yarn 3

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Migrate to yarn 3

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-6880

### CODE REVIEW

**Comment by ibuziuk** [comment]
@SDawley please, review

**Comment by SDawley** [comment]
Dependency fetch worked locally, testing the downstream build now.

**Review by akurinnoy — APPROVED** [review]

**Comment by olexii4** [comment]
@SDawley I have added a script for temporary support and easy switch between Yarn 1 and Yarn 3.

It is possible to build project with different version of package manager:
```sh
${PWD}/scripts/yarn/change_package_manager.sh
```
Знімок екрана 2024-08-29 о 15 27 37

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,13 @@ yarn-error.log
 !.yarn/sdks
 !.yarn/versions
 .pnp.*
+.yarn/cache
+.yarn/install-state.gz
 
 run/.che-dashboard-pod
 run/.custom-resources
-run/public-certs
+run/public-certsg
```
Comment (akurinnoy) [inline .gitignore:20]: typo

```suggestion
run/public-certs
```

**Review by akurinnoy — APPROVED** [review]
LGTM

**Review by SDawley — APPROVED** [review]
Tested and approved

**Pull request merged**